### PR TITLE
Skills to learn

### PR DESCRIPTION
**Pull request by tibetsprague**
Don't really love this solution, but after many attempts this was the easiest/best way i could come up with
Fixes https://github.com/Hylozoic/hylo-node/issues/556

### CODE REVIEW

File: api/graphql/mutations/index.js [hunk 1]
```diff
@@ -147,7 +147,7 @@ export function unlinkAccount (userId, provider) {
   .then(() => ({success: true}))
 }
 
-export async function addSkill (userId, name) {
+export async function addSkill (userId, name, type) {
   name = trim(name)
```
Comment (lorenjohnson) [inline api/graphql/mutations/index.js:151]: Could check for valid type here...

**Comment by lorenjohnson** [comment]
Hey @tibetsprague I've came back and looked at this now a couple times. I continue to be feel it'd be best to keep this flat leaving the Skill `type` handling entirely here on the backend, sticking with `person { skillsToLearn {} }`, and creating an `addSkillToLearn` mutation (vs `addSkill (type)`) in the related graphql schema and resolvers. 

Leaving the abstraction exposed could have its use, but I think as it passes through the layers it's causing more confusion than utility at this stage. 

Thoughts?

**Comment by tibetsprague** [comment]
> Hey @tibetsprague I've came back and looked at this now a couple times. I continue to be feel it'd be best to keep this flat leaving Skill `type` handling entirely here on the backend, sticking with `person { skillsToLearn {} }`, and creating an `addSkillToLearn` mutation (vs `addSkill (type)`) in the related graphql schema and resolvers.
> 
> Leaving the abstraction exposed could have its use, but I think as it passes through the layers it's causing more confusion than utility at this stage.
> 
> Thoughts?

yeah that makes sense for now too, just simplifies a bit more

**Comment by tibetsprague** [comment]
@lorenjohnson ready for new review

File: api/graphql/schema.graphql [hunk 2]
```diff
@@ -869,6 +873,7 @@ type CommunityQuerySet {
 type Skill {
   id: ID
   name: String
+  type: Int
```
Comment (lorenjohnson) [inline api/graphql/schema.graphql:876]: I would keep the `type` out as it's not being used. If a differentiation is needed in type then I'd carry through treating this as a separate type and make a `SkillToLearn` and `SkillsToLearnQuerySet`.

**Review by lorenjohnson — APPROVED** [review]
👍  Overall looks great! 

I do think for clarity it'd be best to leave Skill#type entirely out of the graphql side, leaving it there otherwise may confuse or misdirect a future developer--but that's a nitpick at this point.

**Pull request merged**